### ISSUE TITLE
NEON support

### ISSUE THREAD

**Issue by archimedeschess** [jdart1/arasan-chess#46]
Thank you for the NEON support. I can compile the chess engine with Android NDK and the -DNEON parameter, but I don't see any speed advantages (device is arm64-v8a).

My Android.mk works so far for all devices and looks like this (so you can see which parameters I use when compiling):
```
LOCAL_PATH := $(call my-dir)
include $(CLEAR_VARS)
LOCAL_MODULE := Arasan
LOCAL_C_INCLUDES := ../src/nnue ../src/syzygy/src
LOCAL_SRC_FILES := $(wildcard ../src/*.cpp) ../src/syzygy/src/tbprobe.c
LOCAL_CFLAGS += -std=c99 -DNDEBUG -O3 -flto -fexperimental-new-pass-manager -Wfatal-errors -Wall -Wextra -DNNUE -DSMP -DSMP_STATS -DSYZYGY_TBS -DARASAN_VERSION="230326"
LOCAL_CPPFLAGS += -std=c++17
LOCAL_CPP_FEATURES := exceptions

ifeq ($(TARGET_ARCH_ABI), arm64-v8a)
  LOCAL_CFLAGS += -D_64BIT -DNEON -DUSE_POPCNT
else
  ifeq ($(TARGET_ARCH_ABI), armeabi-v7a)
    LOCAL_CFLAGS += -DNEON
  else
    ifeq ($(TARGET_ARCH_ABI), x86)
      LOCAL_CFLAGS += -mssse3 -mfpmath=sse -m32 -DSIMD -DSSE2 -DUSE_ASM -DUSE_INTRINSICS
    else
      ifeq ($(TARGET_ARCH_ABI), x86_64)
        LOCAL_CFLAGS += -msse4.2 -mpopcnt -m64 -D_64BIT -DSIMD -DSSE2 -DSSE41 -DUSE_ASM -DUSE_INTRINSICS -DUSE_POPCNT
      endif
    endif
  endif
endif

LOCAL_LDFLAGS += 
include $(BUILD_EXECUTABLE)
```

**Comment by jdart1** [comment]
I think your NEON compile is missing -DSIMD.

**Comment by archimedeschess** [comment]
This is intentional. With -DSIMD I get the following error message for arm64-v8a:
```
[arm64-v8a] Compile++      : Arasan <= arasanx.cpp
In file included from /data/data/com.termux/files/home/cecsa/src/Arasan_230326/jni/../src/arasanx.cpp:7:
In file included from /data/data/com.termux/files/home/cecsa/src/Arasan_230326/jni/../src/globals.h:18:
In file included from /data/data/com.termux/files/home/cecsa/src/Arasan_230326/jni/../src/nnue/nnue.h:19:
../src/nnue/simd.h:252:29: fatal error: incompatible pointer types assigning to 'const long *' from 'const long long *'
        outp[i] = vld1q_s64(reinterpret_cast<const long long*>(inp + i));
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/data/com.termux/files/usr/opt/android-ndk/toolchains/llvm/prebuilt/linux-x86_64/lib64/clang/12.0.9/include/arm_neon.h:8630:46: note: expanded from macro 'vld1q_s64'
  __ret = (int64x2_t) __builtin_neon_vld1q_v(__p0, 35); \
                                             ^~~~
1 error generated.
```
-DSIMD only works for x86 and x86_64.

**Comment by jdart1** [comment]
Ok, I will take a look. -DSIMD is supposed to work with NEON.

**Comment by jdart1** [comment]
Fixed in 33f581ff432c0d746bb0e35631dceef3597e9d2b. 
You do not need a custom CMake file. For an NDK build, you can do something like:
'''
cmake -DCMAKE_TOOLCHAIN_FILE=toolchain -DNEON=ON <path to src>
``
where the toolchain file has contents like:
```
set(CMAKE_SYSTEM_NAME Android)
set(CMAKE_SYSTEM_VERSION 21) # API level
set(CMAKE_ANDROID_ARCH_ABI arm64-v8a)
set(CMAKE_ANDROID_NDK /home/jdart/Android/android-ndk-r25c)
```
This requires though a fairly recent cmake version (>=3.21) because NDK toolchain support is lacking in prior versions.
